services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/QinQFlowSpec.groovy

```diff
@@ -0,0 +1,365 @@
+        10        | 20             | 30        | 40
+        10        | 20             | 30        | 0
+        10        | 20             | 0         | 0
+        10        | 0              | 0         | 40
+        0         | 20             | 30        | 0
+    }
+
+    @Tags(TOPOLOGY_DEPENDENT)
+    def "System doesn't allow to create a QinQ flow when a switch doesn't support multi table mode"() {
+        given: "A switch pair with disabled multi table mode on the src switch"
+        def swP = topologyHelper.getAllNeighboringSwitchPairs().find {
+            [it.src, it.dst].any { northbound.getSwitchProperties(it.dpId).multiTable }
+        } ?: assumeTrue("No suiting switches found", false)
+        def initSrcSwProps = northbound.getSwitchProperties(swP.src.dpId)
+        northbound.updateSwitchProperties(swP.src.dpId,
+                northbound.getSwitchProperties(swP.src.dpId).tap { multiTable = false })
+        Wrappers.wait(RULES_INSTALLATION_TIME, 1) {
+            assert northbound.getSwitchRules(swP.src.dpId).flowEntries*.cookie.sort() == swP.src.defaultCookies.sort()
+        }
+        when: "Try to create a QinQ flow when at least on switch doesn't support multi table mode"
+
+        def flow = flowHelperV2.randomFlow(swP)
+        flow.source.innerVlanId = 4093
+        flow.destination.innerVlanId = 3904
+        northboundV2.addFlow(flow)
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+        exc.responseBodyAsString.to(MessageError).errorMessage == "Could not create flow"
+
+        cleanup: "Revert system to original state"
+        northbound.updateSwitchProperties(swP.src.dpId, initSrcSwProps)
+        Wrappers.wait(RULES_INSTALLATION_TIME, 1) {
+            assert northbound.getSwitchRules(swP.src.dpId).flowEntries*.cookie.sort() == swP.src.defaultCookies.sort()
+        }
+    }
+
+    @Ignore("is not implemented yet")
+    @Unroll
+    def "System doesn't allow to create a QinQ flow with incorrect innerVlanIds\
+(src:srcInnerVlanId, dst:dstInnerVlanId)"() {
+        when: "Try to create a QinQ flow with incorrect innerVlanId"
+        def swP = topologyHelper.getAllNeighboringSwitchPairs().find {
+            [it.src, it.dst].any { northbound.getSwitchProperties(it.dpId).multiTable }
+        } ?: assumeTrue("No suiting switches found", false)
+
+        def flow = flowHelperV2.randomFlow(swP)
+        flow.source.innerVlanId = srcInnerVlanId
+        flow.destination.innerVlanId = dstInnerVlanId
+        northboundV2.addFlow(flow)
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 404
+        //TODO(andriidovhan) add error message
+        exc.responseBodyAsString.to(MessageError).errorMessage == "Could not create flow"
+
+        cleanup:
+        flow && flowHelperV2.deleteFlow(flow.flowId)
```

Comment:
I'd rather use
`!exc && flowHelperV2.deleteFlow(flow.flowId)`
in order to only try to delete if no exception was thrown